### ISSUE TITLE
"reim_json" dataset for inference

### ISSUE THREAD

**Issue by JACKTANSNAKE** [surgical-vision/REIM-NeRF#1]
Hi, thank you so much for this work!

I am trying to use your model to generate some novel views for a synthetic dataset. While I was trying to render with trained model, I notice that in https://github.com/surgical-vision/REIM-NeRF/blob/9be61a9700896223ad1a5cfd45067ee375f40184/scripts/inference.py#L26 there are no name for previously trained 'reim_json' dataset. Do I just add this to the options for inference?

Jack

**Comment by dimitrisPs** [comment]
Hi Jack, 

Thank you for reaching out and letting us know about this issue. We apologise for any confusion this may have caused.

We overlooked updating the naming convention for this particular script while refactoring our code for public release. 
We have updated the inference bash and Python scripts to rectify this discrepancy. Furthermore, we now provide a .dev-container  and normalized poses, to help users render from C3VD without having to configure an environment from scratch or download and pre-process C3VD. 

I am leaving this issue open in case you need further assistance. 

Many thank, 
Dimitris

**Comment by JACKTANSNAKE** [comment]
Hi Dimitris,

Thank you so much for the reply and update! In fact I do have issues regarding getting much worse results in rendered scenes than the ones mentioned in your paper. I tried the pretrained weights for dataset desc_t4_a, and reim_json dataset generated following the scripts in the pre-process section and here is what I have for the reconstructed scene using the ls_loc_depth0.03 model: 
![desc_t4_a_trained_weights](https://github.com/surgical-vision/REIM-NeRF/assets/43002751/451ee1d7-8cb2-4517-88ee-5ff38bb508ab) 
and even after training from scratch this is what I get for reconstruction: 
![desc_t4_recon](https://github.com/surgical-vision/REIM-NeRF/assets/43002751/6cff3d69-b398-4182-8edc-e19a843c637c)
Similar with the vanilla NeRF and ls_loc_depth0 models:
![desc_t4_a_nerf](https://github.com/surgical-vision/REIM-NeRF/assets/43002751/2f937247-2de5-464f-912c-4adea497b5cb)
![desc_t4_a_ls_loc_depth0](https://github.com/surgical-vision/REIM-NeRF/assets/43002751/7138c194-e781-4d0a-8c14-1333acb26697)
I am trying to determine what might have gone wrong during the entire process. I have noticed large differences among all of the .json files for training you posted and the ones I generated using the pre-process_c3vd.py. Here is my transforms_test.json generated for the dataset desc_t4_a: 
[transforms_test.json](https://github.com/surgical-vision/REIM-NeRF/files/13397260/transforms_test.json)
The scene scale seems to be very different. I think this is might be caused by this warning during running the pre-processing_c3vd.py, the message is as follows:
(REIM-Nerf) E:\Project\REIM-NeRF>python -m scripts.pre-process_c3vd D:\Data\c3vd\input D:\Data\c3vd --overwrite
saved data under:D:\Data\c3vd\cecum_t1_a
100%|████████████████████████████████████████████████████████████████████████████████| 276/276 [00:43<00:00,  6.32it/s]
E:\Project\REIM-NeRF\reimnerf\datasets\preprocessing\raw_data.py:253: RuntimeWarning: invalid value encountered in cast
  distmap = ((distmap/distmap_scale)*(2**16-1)).astype(np.uint16)
saved data under:D:\Data\c3vd\desc_t4_a
100%|████████████████████████████████████████████████████████████████████████████████| 148/148 [00:25<00:00,  5.84it/s]
The pre-processing finishes after the warning but the dataset is created just fine. For some scenes this warning pops up(e.g. cecum_t1_a) but for some other scenes it doesn't(e.g. desc_t4_a). Do you think this could the be the problem? Also in your instructions for pre-processing, I don't see that the pose.txt is mentioned in the tree diagram. Are you using the pose.txt directly from the c3vd datasets or do the poses need some additional pre-processing?
I will try the poses that you guys just posted for reconstructing the scenes. Thank you so much again for this work and I apologize for raising so many questions at the same time!

Best
Jack

**Comment by JACKTANSNAKE** [comment]
Hi, just some more updates. I was able to render very good scenes using the poses in the repo. 
![desc_t4_a](https://github.com/surgical-vision/REIM-NeRF/assets/43002751/f0b0ef43-7477-457d-b758-978d7ffbf728)
Another one for the cecum_t1_a dataset:
![cecum_t1_a](https://github.com/surgical-vision/REIM-NeRF/assets/43002751/da03fa0e-59c8-4f6c-967a-cf3aa5def6e6)
I think the problem is somewhere within the normalizing part in raw_data.py, but I can't figure out what is causing this problem. Could you check if there is anything that might have led to this?
FYI, when I was pre-processing the data, I only modified this line of path: https://github.com/surgical-vision/REIM-NeRF/blob/fab23de8f5caf66dc35f543f41b469e46f55d756/reimnerf/datasets/preprocessing/raw_data.py#L266C9-L266C9
to be my local absolute path so my machine could properly read the calibration file. I did not modify any other code in that file or the pre-process_c3vd.py, so the other parts of that piece of code should be exactly the same as the one in the repo. The environment that the pre-processing was performed is Windows 10 with python 3.10, and all packages in the requirements.txt have been installed as required version. 
I was trying to retrain from scratch by replacing the poses in my data with yours, but I realized that the distmaps and depthmaps are also rescaled during the normalization process using the same scales. So I guess it does not make sense to do it without figuring out which part I messed up. 

Thank you so much again and hope to hear back from you soon!

Best
Jack

**Comment by dimitrisPs** [comment]
Hi Jack,

Thank you for the feedback, it is nice to know that you could render nice-looking scenes using the provided models and poses. I will do my best to answer your questions below

### What is causing the issue
It seems C3VD implemented updates since we used it to develop our paper. One of those altered the format of the pose.txt file which we used to reconstruct and normalize each scene. In the past, we had to read the pose file from .txt using numpy and then reshape the provided vectors to arrays. The new format requires us to additionally transpose the pose arrays.  

### How to fix the issue 
Changes have been implemented in commit (711f571) together with a fix to the calibration camera path. Please download the latest copy.

### How well do the new scripts work with the updated dataset?
We tested the output of the updated scripts with two sequences and the resulting poses seem to be the same up to 1e-5. We also tested rendering, using poses generated from the new version of C3VD with the provided models (old dataset) and qualitative results seem to be very similar. 

### What else has changed?
We also noticed that the C3VD authors changed the naming convention of the dataset and therefore the names of the provided pose files and pre-trained model will be incompatible with the current C3VD distribution (for instance t1v1 is now named trans_t1_a). This means that processing the data yourself will result in dataset names that are incompatible with the provided bash scripts and the pre-trained models. If however, you process C3VD and train models using the provided scripts from scratch, the naming should be consistent with the new C3VD version. 

We are not sure of what else has changed in C3VD and whether or not numerical results can be recreated using the new version of the dataset but we believe relative accuracy gains as presented in the ablation study of the paper should hold.

### Run-time warning when saving dist-maps

The warning you get typically means that we are performing conventions from floats to uint16 of a nan value or a value that is out of the uint16 range. With the updated script and latest version of C3VD, we didn't observe any such warning while processing  cecum_t1_a.

### How pre-processing uses pose.txt

The pose.txt contains the provided C3VD poses. The pre-processing scrips read those poses and transform their corresponding pose point cloud w.r.t. the world frame. point clouds from all poses are merged and the final point cloud is used to understand the scale of the scene and normalize all poses. 

We really appreciate you taking the time to describe the issue you encountered here. Without this process, we wouldn't be aware of the changes in C3VD and the provided scripts would not work for the rest of the community.

Please feel free to reach out again. 

Many thanks, 
Dimitris

**Comment by JACKTANSNAKE** [comment]
Thanks for the lightning fast reply! Much appreciated!

**Comment by dimitrisPs** [comment]
Hi Jack, 

Do you need any further help with this? If not, I will close this issue

**Comment by JACKTANSNAKE** [comment]
Hi Dimitris,

I am fine for the moment. Thank you so much for your help. I will open a new issue if I have further questions. Much appreciated!
